lib/models/subscription.js

```diff
@@ -173,7 +190,12 @@ module.exports = (sequelize, DataTypes) => {
     },
 
     async unsubscribe(githubId, channelId, slackWorkspaceId) {
-      await this.destroy({ where: { githubId, channelId, slackWorkspaceId } });
+      const where = { githubId, channelId, slackWorkspaceId };
+      const subscriptions = await this.findAll({ where });
+      await this.destroy({ where, individualHooks: true });
+
+      const ids = subscriptions.map(subscription => subscription.id);
+      await sequelize.model('DeletedSubscription').update({ reason: 'unsubscribe' }, { where: { id: ids } });
```

Comment:
it'd be cool if we could pass a reason + multiple subscriptions into `destroyWithReason`, otherwise this could be a bit race-conditiony?

Reply:
I'm not doing that because I wanted to keep the bulk delete. I think this is very covered by tests so I would keep it like it is.